Application/src/view/visuals/component/ClassRelationship.java

```diff
@@ -0,0 +1,249 @@
+package view.visuals.component;
+
+import javafx.scene.SnapshotParameters;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.paint.Color;
+import view.visuals.Renderable;
+
+/**
+ * Class for simulating the relationships of the class diagram
+ *
+ * @author Rashad Kamsheh
+ * @version 1.1
+ */
+
+public class ClassRelationship implements Renderable {
+
+
+    //Image for the road sprite which are used to resemble the class diagram relationships
+    private static Image road = new Image("resources/roadSprite.png");
+    //ImageView that contains the arrow, used ImageView to be able to rotate
+    private static ImageView arrow = new ImageView("resources/road-arrow.png");
+    private Coordinates coordinates; //coordinates
+    private Coordinates fromNode, toNode;
+    private int size; //size of the allowed class space
+
+    // constructor
+    public ClassRelationship(Coordinates fromNode, Coordinates toNode, int size) {
+        this.fromNode = fromNode;
+        this.toNode = toNode;
+        this.size = size;
+    }
+
+    // initialiser for rendering the road sprites between a super class and a sub class
+    public void init(Coordinates fromNode, Coordinates toNode, int size) {
+        this.fromNode = fromNode;
+        this.toNode = toNode;
+        this.size = size;
+    }
+
+    @Override
+    public void render(GraphicsContext gc) {
```

Comment:
160 line method? I think a recommendation is no more than 25 lines excluding comments.